### PR TITLE
Update linters & checkers

### PR DESCRIPTION
**Pull request by fmauch**
Since the main branch is targeted for ubuntu jammy, where there is no
clang-format-10 available anymore, we should raise that to 14, as this is
the default clang-format version on ubuntu jammy.

### CODE REVIEW

**Comment by fmauch** [comment]
Since this PR became rather large I would be fine with rebase-merging this.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -65,59 +64,16 @@ repos:
...
-  # Copyright
-  - repo: local
-    hooks:
-      - id: ament_copyright
-        name: ament_copyright
-        description: Check if copyright notice is available in all files.
-        stages: [commit]
-        entry: ament_copyright
-        language: system
-
```
Comment (RobertWilbrandt) [inline .pre-commit-config.yaml:120]: I don't really agree with removing these two for the sake of runtime reduction. I think they actually check for problems that are easily introduced by mistake and if we don't want the duplication in the ci we should tackle it there.

Comment (fmauch) [reply .pre-commit-config.yaml:120]: That is definitively a fair point.

Comment (fmauch) [reply .pre-commit-config.yaml]: I turned it the other way around in 85558b1

**Review by RobertWilbrandt — APPROVED** [review]
Consolidating the workflows seems like a good move. Thanks!

**Pull request merged**